### PR TITLE
feat(accordion): add methods to expand/collapse/collapseAll panels

### PR DESCRIPTION
**Pull request by Thorski**
### feat(accordion): add methods to expand/collapse/collapseAll panels

Added
  expand, collapse, collapseAll, and isExpanded methods.

#1970

### CODE REVIEW

**Comment by Thorski** [comment]
What needs to be done to get this into a build?   

Thanks.

**Comment by christarczon** [comment]
Hopefully this is accepted. Can you add expandAll() as well?

**Comment by Thorski** [comment]
@christarczon I don't know if I have missed a step or why this has not been accepted.   I wish someone would please advise, I would like to use this from an official release.   An expandAll could be added easily.

File: src/accordion/accordion.ts [hunk 1]
```diff
@@ -175,6 +175,44 @@ export class NgbAccordion implements AfterContentChecked {
...
   }
 
+  /**
+   * Programmatically expand a panel with a given id if it is closed.
+   */
+  expand(panelId: string) {
+    const panel = this.panels.find(p => p.id === panelId);
+
+    if (panel && !panel.disabled && !panel.isOpen) {
+      this.toggle(panelId);
```
Comment (pkozlowski-opensource) [inline src/accordion/accordion.ts]: IMO we should it differently - the logic of expanding should be in this method and `toggle` should call it based on the state. In other words - the `toggle` method should make use of the new `expand` / `collapse ` and not the other way around.

File: src/accordion/accordion.ts [hunk 2]
```diff
@@ -175,6 +175,44 @@ export class NgbAccordion implements AfterContentChecked {
...
+    this.panels.forEach(panel => { panel.isOpen = false; });
+    this._updateActiveIds();
+  }
+
+  /**
+   * Returns if a panel is currently expanded.
+   */
+  isExpanded(panelId): boolean {
+    const panel = this.panels.find(p => p.id === panelId);
+    return panel.isOpen;
```
Comment (pkozlowski-opensource) [inline src/accordion/accordion.ts]: This will crash if there is no panel with a given `panelId` if I'm not mistaken? Could you please add a test for this case and fix the code if needed?

File: src/accordion/accordion.ts [hunk 3]
```diff
@@ -175,6 +175,44 @@ export class NgbAccordion implements AfterContentChecked {
...
+    if (panel && !panel.disabled && panel.isOpen) {
+      this.toggle(panelId);
+    }
+  }
+
+  /**
+   * Programmatically collapse all panels.
+   */
+  collapseAll() {
+    this.panels.forEach(panel => { panel.isOpen = false; });
```
Comment (pkozlowski-opensource) [inline src/accordion/accordion.ts]: This should rather delegate to the `collapse` call - otherwise we are not taking the `disabled` flag into account. Please add a test for the case assuring that we are not collapsing an open, disabled panel.

**Comment by pkozlowski-opensource** [comment]
@Thorski sorry for the delay reviewing this PR - I've finally got to it and left few comments. Generally speaking:
* the implementation is "light" on tests and might be missing some use-cases;
* I would approach implementation differently to make it simpler and make sure that we are not missing corner cases.

Additionally I would like to propose that we do the following renaming:
* `expand` => `open`
* `expandAll` => `openAll`
* `collapse` => `close`
* `collapseAll` => closeAll
* `isExpanded` => `isOpen`

This would align names with other directives in this library. WDYT?

File: src/accordion/accordion.ts (hunk 3 above)
Comment (Thorski) [reply src/accordion/accordion.ts]: I wasn't sure if the disabled flag should take effect here or not, because the 'closeOtherPanels' option I believe takes effect even when a panel is disabled (see toggle and _closeOthers).

Comment (Thorski) [reply src/accordion/accordion.ts]: Also, I am not sure how to make a panel in the state of 'open and disabled'.   Is it possible?  If I set disabled on a panel, it closes.   How do I do this?

**Comment by pkozlowski-opensource** [comment]
This PR doesn't merge cleanly, is very light on tests and has several review comments not addressed so we are going to work towards merging #2595 instead.
